Spongeforge Crash

```
### GregTech CEu Version

2.5.0-beta

### Modpack Used

 Nomifactory (GTCEu Port)

### Addons Installed

Spongeforge 1.12.2-2838-7.4.8-RC4142

### Environment

Sponge Server

### Side Crashed

Server

### New Worlds

Unsure

### Steps to Reproduce

Make a server with Nomifactory (GTCEu Port) modpack and run with spongeforge

### Crash Report

---- Minecraft Crash Report ----
// Lolis deobfuscated this stacktrace using MCP's stable-39 mappings.
// Shall we play a game?

Time: 2023-02-24 08:31:13 CET
Description: Exception in server tick loop

```
net.minecraftforge.fml.common.LoaderExceptionModCrash: Caught exception from GregTech (gregtech)
Caused by: java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: gregtech/api/items/toolitem/ItemGTSword
    at gregtech.api.items.toolitem.ItemGTSword$Builder.supply(ItemGTSword.java:302)
    at gregtech.api.items.toolitem.ToolBuilder.build(ToolBuilder.java:101)
    at gregtech.common.items.ToolItems.register(ToolItems.java:320)
    at gregtech.common.items.ToolItems.init(ToolItems.java:63)
    at gregtech.core.CoreModule.preInit(CoreModule.java:149)
    at gregtech.modules.ModuleManager.onPreInit(ModuleManager.java:104)
    at gregtech.GregTechMod.preInit(GregTechMod.java:50)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at net.minecraftforge.fml.common.FMLModContainer.handleModStateEvent(FMLModContainer.java:637)
    at sun.reflect.GeneratedMethodAccessor9.invoke(Unknown Source)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
    at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
    at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
    at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
    at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
    at com.google.common.eventbus.EventBus.post(EventBus.java:217)
    at net.minecraftforge.fml.common.LoadController.sendEventToModContainer(LoadController.java:219)
    at net.minecraftforge.fml.common.LoadController.propogateStateMessage(LoadController.java:197)
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:498)
    at com.google.common.eventbus.Subscriber.invokeSubscriberMethod(Subscriber.java:91)
    at com.google.common.eventbus.Subscriber$SynchronizedSubscriber.invokeSubscriberMethod(Subscriber.java:150)
    at com.google.common.eventbus.Subscriber$1.run(Subscriber.java:76)
    at com.google.common.util.concurrent.MoreExecutors$DirectExecutor.execute(MoreExecutors.java:399)
    at com.google.common.eventbus.Subscriber.dispatchEvent(Subscriber.java:71)
    at com.google.common.eventbus.Dispatcher$PerThreadQueuedDispatcher.dispatch(Dispatcher.java:116)
    at com.google.common.eventbus.EventBus.post(EventBus.java:217)
    at net.minecraftforge.fml.common.LoadController.redirect$zza000$forgeImpl$PostEvent(LoadController.java:1068)
    at net.minecraftforge.fml.common.LoadController.distributeStateMessage(LoadController.java:136)
    at net.minecraftforge.fml.common.Loader.preinitializeMods(Loader.java:629)
    at net.minecraftforge.fml.server.FMLServerHandler.beginServerLoading(FMLServerHandler.java:99)
    at net.minecraftforge.fml.common.FMLCommonHandler.onServerStart(FMLCommonHandler.java:333)
    at net.minecraft.server.dedicated.DedicatedServer.init(DedicatedServer.java:125)
    at net.minecraft.server.MinecraftServer.run(MinecraftServer.java:486)
    at java.lang.Thread.run(Thread.java:750)
Caused by: java.lang.NoClassDefFoundError: gregtech/api/items/toolitem/ItemGTSword
    ... 43 more
Caused by: java.lang.ClassNotFoundException: gregtech.api.items.toolitem.ItemGTSword
    at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:191)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:419)
    at java.lang.ClassLoader.loadClass(ClassLoader.java:352)
    ... 43 more
Caused by: java.lang.VerifyError: class gregtech.api.items.toolitem.ItemGTSword overrides final method getId.()Ljava/lang/String;
    at java.lang.ClassLoader.defineClass1(Native Method)
    at java.lang.ClassLoader.defineClass(ClassLoader.java:757)
    at java.security.SecureClassLoader.defineClass(SecureClassLoader.java:142)
    at net.minecraft.launchwrapper.LaunchClassLoader.findClass(LaunchClassLoader.java:182)
    ... 45 more
```


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Sponge PhaseTracker --
  Phase Stack: [Empty stack]

-- System Details --
  Minecraft Version: 1.12.2
  Operating System: Windows 10 (amd64) version 10.0
  Java Version: 1.8.0_362, Azul Systems, Inc.
  Java VM Version: OpenJDK 64-Bit Server VM (mixed mode), Azul Systems, Inc.
  Memory: 473159880 bytes (451 MB) / 1509949440 bytes (1440 MB) up to 8589934592 bytes (8192 MB)
  JVM Flags: 19 total; -Xmx8G -XX:+UseG1GC -XX:+ParallelRefProcEnabled -XX:MaxGCPauseMillis=200 -XX:+UnlockExperimentalVMOptions -XX:+DisableExplicitGC -XX:+AlwaysPreTouch -XX:G1NewSizePercent=30 -XX:G1MaxNewSizePercent=40 -XX:G1HeapRegionSize=8M -XX:G1ReservePercent=20 -XX:G1HeapWastePercent=5 -XX:G1MixedGCCountTarget=4 -XX:InitiatingHeapOccupancyPercent=15 -XX:G1MixedGCLiveThresholdPercent=90 -XX:G1RSetUpdatingPauseTimePercent=5 -XX:SurvivorRatio=32 -XX:+PerfDisableSharedMem -XX:MaxTenuringThreshold=1
  IntCache: cache: 0, tcache: 0, allocated: 0, tallocated: 0
  FML: MCP 9.42 LoliASM 5.6 Powered by Forge 14.23.5.2860 163 mods loaded, 163 mods active
       States: 'U' = Unloaded 'L' = Loaded 'C' = Constructed 'H' = Pre-initialized 'I' = Initialized 'J' = Post-initialized 'A' = Available 'D' = Disabled 'E' = Errored
       
       Mods:
       
       | State | ID                                | Version                             | Source                                                      | Signature                                |
       |:----- |:--------------------------------- |:----------------------------------- |:----------------------------------------------------------- |:---------------------------------------- |
       | LCH   | minecraft                         | 1.12.2                              | minecraft.jar                                               | None                                     |
       | LCH   | mcp                               | 9.42                                | minecraft.jar                                               | None                                     |
       | LCH   | bansoukou                         | 4.3.1                               | minecraft.jar                                               | None                                     |
       | LCH   | FML                               | 8.0.99.99                           | forge-1.12.2-14.23.5.2860.jar                               | e3c3d50c7c986df74c645c0ac54639741c90a557 |
       | LCH   | forge                             | 14.23.5.2860                        | forge-1.12.2-14.23.5.2860.jar                               | e3c3d50c7c986df74c645c0ac54639741c90a557 |
       | LCH   | advancedrocketrycore              | 1                                   | minecraft.jar                                               | None                                     |
       | LCH   | creativecoredummy                 | 1.0.0                               | minecraft.jar                                               | None                                     |
       | LCH   | littletilescore                   | 1.0.0                               | minecraft.jar                                               | None                                     |
       | LCH   | mixinbooter                       | 4.2                                 | minecraft.jar                                               | None                                     |
       | LCH   | spongeapi                         | 7.4.0-5ea6215                       | aaa_spongeforge-1.12.2-2838-7.4.8-RC4142.jar                | None                                     |
       | LCH   | sponge                            | 1.12.2-7.4.8-SNAPSHOT               | aaa_spongeforge-1.12.2-2838-7.4.8-RC4142.jar                | None                                     |
       | LCH   | spongeforge                       | 1.12.2-2838-7.4.8-RC4142            | aaa_spongeforge-1.12.2-2838-7.4.8-RC4142.jar                | None                                     |
       | LCH   | foamfixcore                       | 7.7.4                               | minecraft.jar                                               | None                                     |
       | LCH   | tickcentral                       | 3.2                                 | TickCentral-3.2.jar                                         | None                                     |
       | LCH   | randompatches                     | 1.12.2-1.22.1.10                    | randompatches-1.12.2-1.22.1.10.jar                          | None                                     |
       | LCH   | fastbench                         | 1.7.3                               | FastWorkbench-1.12.2-1.7.3.jar                              | None                                     |
       | LCH   | actuallyadditions                 | 1.12.2-r152                         | ActuallyAdditions-1.12.2-r152.jar                           | None                                     |
       | LCH   | baubles                           | 1.5.2                               | Baubles-1.12-1.5.2.jar                                      | None                                     |
       | LCH   | actuallybaubles                   | 1.1                                 | ActuallyBaubles-1.12-1.1.jar                                | None                                     |
       | LCH   | redstoneflux                      | 2.1.1                               | RedstoneFlux-1.12-2.1.1.1-universal.jar                     | None                                     |
       | LCH   | cofhcore                          | 4.6.6                               | CoFHCore-1.12.2-4.6.6.1-universal.jar                       | None                                     |
       | LCH   | libvulpes                         | 0.4.2.-75                           | LibVulpes-1.12.2-0.4.2-75-universal.jar                     | None                                     |
       | LCH   | advancedrocketry                  | 1.7.0.-232                          | AdvancedRocketry-1.12.2-1.7.0-232-universal.jar             | None                                     |
       | LCH   | advsolars                         | 1.0                                 | advsolars-1.0.jar                                           | None                                     |
       | LCH   | appliedenergistics2               | rv6-stable-7-extended_life-v0.54.19 | appliedenergistics2-rv6-stable-7-extended_life-v0.54.19.jar | None                                     |
       | LCH   | bdlib                             | 1.14.3.12                           | bdlib-1.14.3.12-mc1.12.2.jar                                | None                                     |
       | LCH   | ae2stuff                          | 0.7.0.4                             | ae2stuff-0.7.0.4-mc1.12.2.jar                               | None                                     |
       | LCH   | endercore                         | 1.12.2-0.5.76                       | EnderCore-1.12.2-0.5.76.jar                                 | None                                     |
       | LCH   | crafttweaker                      | 4.1.20                              | CraftTweaker2-1.12-4.1.20.673.jar                           | None                                     |
       | LCH   | mtlib                             | 3.0.7                               | MTLib-3.0.7.jar                                             | None                                     |
       | LCH   | modtweaker                        | 4.0.19                              | modtweaker-4.0.20.5.jar                                     | None                                     |
       | LCH   | jei                               | 4.23.0                              | HadEnoughItems_1.12.2-4.23.0.jar                            | None                                     |
       | LCH   | codechickenlib                    | 3.2.3.358                           | CodeChickenLib-1.12.2-3.2.3.358-universal.jar               | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
       | LCH   | brandonscore                      | 2.4.20                              | BrandonsCore-1.12.2-2.4.20.162-universal.jar                | None                                     |
       | LCH   | cofhworld                         | 1.4.0                               | CoFHWorld-1.12.2-1.4.0.1-universal.jar                      | None                                     |
       | LCH   | thermalfoundation                 | 2.6.7                               | ThermalFoundation-1.12.2-2.6.7.1-universal.jar              | None                                     |
       | LCH   | draconicevolution                 | 2.3.28                              | Draconic-Evolution-1.12.2-2.3.28.354-universal.jar          | None                                     |
       | LCH   | thermalexpansion                  | 5.5.7                               | ThermalExpansion-1.12.2-5.5.7.1-universal.jar               | None                                     |
       | LCH   | enderio                           | 5.3.70                              | EnderIO-1.12.2-5.3.70.jar                                   | None                                     |
       | LCH   | p455w0rdslib                      | 2.3.161                             | p455w0rdslib-1.12.2-2.3.161.jar                             | 186bc454cd122c9c2f1aa4f95611254bcc543363 |
       | LCH   | ae2wtlib                          | 1.0.34                              | AE2WTLib-1.12.2-1.0.34.jar                                  | 186bc454cd122c9c2f1aa4f95611254bcc543363 |
       | LCH   | appleskin                         | 1.0.14                              | AppleSkin-mc1.12-1.0.14.jar                                 | None                                     |
       | LCH   | architecturecraft                 | @VERSION@                           | architecturecraft-1.12-3.98.jar                             | None                                     |
       | LCH   | thedragonlib                      | 1.12.2-5.3.0                        | thedragonlib-1.12.2-5.3.0.jar                               | None                                     |
       | LCH   | armorplus                         | 1.12.2-11.25.1.62                   | armorplus-1.12.2-11.25.1.62.jar                             | None                                     |
       | LCH   | attributefix                      | 1.0.4                               | AttributeFix-1.12.2-1.0.4.jar                               | d476d1b22b218a10d845928d1665d45fce301b27 |
       | LCH   | autoreglib                        | 1.3-32                              | AutoRegLib-1.3-32.jar                                       | None                                     |
       | LCH   | avaritia                          | 3.3.0                               | Avaritia-1.12.2-3.3.0.37-universal.jar                      | None                                     |
       | LCH   | base                              | 3.14.0                              | base-1.12.2-3.14.0.jar                                      | None                                     |
       | LCH   | betterquesting                    | 4.1.0                               | BetterQuestingUnofficial-4.1.0.jar                          | None                                     |
       | LCH   | bookshelf                         | 2.3.590                             | Bookshelf-1.12.2-2.3.590.jar                                | d476d1b22b218a10d845928d1665d45fce301b27 |
       | LCH   | forgelin                          | 1.8.4                               | Forgelin-1.8.4.jar                                          | None                                     |
       | LCH   | bountiful                         | 2.2.2                               | Bountiful-2.2.2.jar                                         | None                                     |
       | LCH   | buildinggadgets                   | 2.8.4                               | BuildingGadgets-2.8.4.jar                                   | None                                     |
       | LCH   | spark                             | 1.5.2                               | spark-forge1122.jar                                         | None                                     |
       | LCH   | loliasm                           | 5.6                                 | censoredasm5.6.jar                                          | None                                     |
       | LCH   | chameleon                         | 1.12-4.1.3                          | Chameleon-1.12-4.1.3.jar                                    | None                                     |
       | LCH   | chesttransporter                  | 2.8.8                               | ChestTransporter-1.12.2-2.8.8.jar                           | None                                     |
       | LCH   | chisel                            | MC1.12.2-1.0.2.45                   | Chisel-MC1.12.2-1.0.2.45.jar                                | None                                     |
       | LCH   | clumps                            | 3.1.2                               | Clumps-3.1.2.jar                                            | None                                     |
       | LCH   | collective                        | 3.0                                 | collective-1.12.2-3.0.jar                                   | None                                     |
       | LCH   | contenttweaker                    | 1.12.2-4.10.0                       | ContentTweaker-1.12.2-4.10.0.jar                            | None                                     |
       | LCH   | craftelytra                       | 1.2                                 | CraftableElytra-1.1.jar                                     | None                                     |
       | LCH   | cns                               | 1.0                                 | CraftingNetherStar 1.12.1 -1.1.jar                          | None                                     |
       | LCH   | craftingtweaks                    | 9.0.1                               | CraftingTweaks_1.12.2-9.0.1.jar                             | None                                     |
       | LCH   | crafttweakerjei                   | 2.0.3                               | CraftTweaker2-1.12-4.1.20.673.jar                           | None                                     |
       | LCH   | creativecore                      | 1.10.0                              | CreativeCore_v1.10.69_mc1.12.2.jar                          | None                                     |
       | LCH   | cucumber                          | 1.1.3                               | Cucumber-1.12.2-1.1.3.jar                                   | None                                     |
       | LCH   | darkutils                         | 1.8.230                             | DarkUtils-1.12.2-1.8.230.jar                                | d476d1b22b218a10d845928d1665d45fce301b27 |
       | LCH   | patchouli                         | 1.0-23.6                            | Patchouli-1.0-23.6.jar                                      | None                                     |
       | LCH   | deepmoblearning                   | 1.0.0                               | DeepMobEvolution-1.1.2.jar                                  | None                                     |
       | LCH   | defaultworldgenerator-port        | 1.12-2.3.1                          | DefaultWorldGenerator-port-1.12-2.3.1.jar                   | None                                     |
       | LCE   | gregtech                          | 2.5.2-beta                          | gregtech-1.12.2-2.5.2-beta.jar                              | None                                     |
       | LC    | devtech                           | 2.1.2                               | devtech-2.1.2.jar                                           | None                                     |
       | LC    | diethopper                        | 1.1                                 | diethopper-1.1.jar                                          | None                                     |
       | LC    | difficultylock                    | 1.2                                 | difficultylock_1.12.2-1.2.jar                               | None                                     |
       | LC    | theoneprobe                       | 1.4.28                              | theoneprobe-1.12-1.4.28.jar                                 | None                                     |
       | LC    | dimensionaledibles                | 1.12.2-1.6.4                        | DimensionalEdibles-1.12.2-1.6.4.jar                         | None                                     |
       | LC    | effortlessbuilding                | 1.12.2-2.16                         | effortlessbuilding-1.12.2-2.16.jar                          | None                                     |
       | LC    | elevatorid                        | 1.3.14                              | ElevatorMod-1.12.2-1.3.14.jar                               | None                                     |
       | LC    | enderiointegrationtic             | 5.3.70                              | EnderIO-1.12.2-5.3.70.jar                                   | None                                     |
       | LC    | enderiobase                       | 5.3.70                              | EnderIO-1.12.2-5.3.70.jar                                   | None                                     |
       | LC    | enderioconduits                   | 5.3.70                              | EnderIO-1.12.2-5.3.70.jar                                   | None                                     |
       | LC    | enderioconduitsappliedenergistics | 5.3.70                              | EnderIO-1.12.2-5.3.70.jar                                   | None                                     |
       | LC    | enderioconduitsopencomputers      | 5.3.70                              | EnderIO-1.12.2-5.3.70.jar                                   | None                                     |
       | LC    | enderioconduitsrefinedstorage     | 5.3.70                              | EnderIO-1.12.2-5.3.70.jar                                   | None                                     |
       | LC    | enderiointegrationforestry        | 5.3.70                              | EnderIO-1.12.2-5.3.70.jar                                   | None                                     |
       | LC    | enderiointegrationticlate         | 5.3.70                              | EnderIO-1.12.2-5.3.70.jar                                   | None                                     |
       | LC    | enderioinvpanel                   | 5.3.70                              | EnderIO-1.12.2-5.3.70.jar                                   | None                                     |
       | LC    | ftblib                            | 5.4.7.2                             | FTBLib-5.4.7.2.jar                                          | None                                     |
       | LC    | enderiomachines                   | 5.3.70                              | EnderIO-1.12.2-5.3.70.jar                                   | None                                     |
       | LC    | enderiopowertools                 | 5.3.70                              | EnderIO-1.12.2-5.3.70.jar                                   | None                                     |
       | LC    | enderioendergy                    | 5.3.70                              | EnderIO-endergy-1.12.2-5.3.70.jar                           | None                                     |
       | LC    | enderstorage                      | 2.4.6.137                           | EnderStorage-1.12.2-2.4.6.137-universal.jar                 | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
       | LC    | extendedcrafting                  | 1.7.0                               | ExtendedCrafting-Nomifactory-Edition-1.7.0.jar              | None                                     |
       | LC    | extrautils2                       | 1.0                                 | extrautils2-1.12-1.9.9.jar                                  | None                                     |
       | LC    | fasterladderclimbing              | 0.1-146                             | FasterLadderClimbing-1.12-0.1-146.jar                       | None                                     |
       | LC    | fastleafdecay                     | v14                                 | FastLeafDecay-v14.jar                                       | None                                     |
       | LC    | foamfix                           | 0.10.14-1.12.2                      | foamfix-0.10.14-1.12.2.jar                                  | None                                     |
       | LC    | forgemultipartcbe                 | 2.6.2.83                            | ForgeMultipart-1.12.2-2.6.2.83-universal.jar                | f1850c39b2516232a2108a7bd84d1cb5df93b261 |
       | LC    | microblockcbe                     | 2.6.2.83                            | ForgeMultipart-1.12.2-2.6.2.83-universal.jar                | None                                     |
       | LC    | minecraftmultipartcbe             | 2.6.2.83                            | ForgeMultipart-1.12.2-2.6.2.83-universal.jar                | None                                     |
       | LC    | forgivingvoid                     | 1.1.0                               | ForgivingVoid_1.12.2-1.1.0.jar                              | None                                     |
       | LC    | storagedrawers                    | 5.2.2                               | StorageDrawers-1.12.2-5.4.2.jar                             | None                                     |
       | LC    | framedcompactdrawers              | 1.2.7                               | framedcompactdrawers-1.2.7.jar                              | None                                     |
       | LC    | ftbbackups                        | 1.1.0.1                             | FTBBackups-1.1.0.1.jar                                      | None                                     |
       | LC    | ftbutilities                      | 5.4.1.131                           | FTBUtilities-5.4.1.131.jar                                  | None                                     |
       | LC    | gcym                              | 1.1.6                               | GregicalityMultiblocks-1.12.2-1.1.6.jar                     | None                                     |
       | LC    | librarianlib                      | 4.22                                | librarianlib-1.12.2-4.22.jar                                | None                                     |
       | LC    | hooked                            | %VERSION%                           | hooked-1.0.3.jar                                            | None                                     |
       | LC    | inventorytweaks                   | 1.63+release.109.220f184            | InventoryTweaks-1.63.jar                                    | 55d2cd4f5f0961410bf7b91ef6c6bf00a766dcbe |
       | LC    | journeymap                        | 1.12.2-5.7.1                        | journeymap-1.12.2-5.7.1.jar                                 | None                                     |
       | LC    | jecalculation                     | 1.12.2-3.2.7                        | JustEnoughCalculation-1.12.2-3.2.7.jar                      | None                                     |
       | LC    | justenoughdimensions              | 1.6.0-dev.20200416.184714           | justenoughdimensions-1.12.2-1.6.0-dev.20200416.184714.jar   | 2b03e1423915a189b8094816baa18f239d576dff |
       | LC    | laggoggles                        | 1.12.2-5.8-132                      | LagGoggles-1.12.2-5.8-132.jar                               | None                                     |
       | LC    | littletiles                       | 1.5.0                               | LittleTiles_v1.5.57_mc1.12.2.jar                            | None                                     |
       | LC    | lostcities                        | 2.0.22                              | lostcities-1.12-2.0.22.jar                                  | None                                     |
       | LC    | morefurnaces                      | 1.11.0                              | MoreFurnaces-Nomifactory-Edition-1.12.2-1.11.0.jar          | None                                     |
       | LC    | morelibs                          | 1.2.2                               | morelibs-1.12.2-1.2.2.jar                                   | None                                     |
       | LC    | mrtjpcore                         | 2.1.4.43                            | MrTJPCore-1.12.2-2.1.4.43-universal.jar                     | None                                     |
       | LC    | projectred-core                   | 4.9.4.120                           | ProjectRed-1.12.2-4.9.4.120-Base.jar                        | None                                     |
       | LC    | moreplates                        | 5.5.2                               | moreplates-1.12.2-5.5.2.jar                                 | None                                     |
       | LC    | morpheus                          | 1.12.2-3.5.106                      | Morpheus-1.12.2-3.5.106.jar                                 | None                                     |
       | LC    | multiblocktweaker                 | 1.12.2-stable-gtceu-1.8.2           | multiblocktweaker-1.12.2-stable-gtceu-1.8.2.jar             | None                                     |
       | LC    | nonetherportals                   | 1.0.0                               | nonetherPortals_v1.0.0formc1.12.2.jar                       | None                                     |
       | LC    | nuclearcraft                      | 2.18y                               | NuclearCraft-2.18y-1.12.2-patched.jar                       | None                                     |
       | LC    | openglider                        | @VERSION@                           | OpenGlider-1.12.1-1.1.0.jar                                 | None                                     |
       | LC    | packagedauto                      | 1.12.2-1.0.5.19                     | PackagedAuto-1.12.2-1.0.5.19.jar                            | None                                     |
       | LC    | packagedexcrafting                | 1.12.2-1.0.2.3                      | PackagedExCrafting-1.12.2-1.0.2.3.jar                       | None                                     |
       | LC    | packmode                          | 1.2.0                               | packmode-1.12.2-1.2.0.jar                                   | None                                     |
       | LC    | placebo                           | 1.6.0                               | Placebo-1.12.2-1.6.0.jar                                    | None                                     |
       | LC    | projectred-integration            | 4.9.4.120                           | ProjectRed-1.12.2-4.9.4.120-integration.jar                 | None                                     |
       | LC    | projectred-transmission           | 4.9.4.120                           | ProjectRed-1.12.2-4.9.4.120-integration.jar                 | None                                     |
       | LC    | redstonearsenal                   | 2.6.6                               | RedstoneArsenal-1.12.2-2.6.6.1-universal.jar                | None                                     |
       | LC    | simplyjetpacks                    | 2.2.15.69                           | SimplyJetpacks2-1.12.2-2.2.15.69.jar                        | None                                     |
       | LC    | snad                              | 1.12.1-1.7.09.16a                   | Snad-1.12.1-1.7.09.16a.jar                                  | None                                     |
       | LC    | solarflux                         | 12.4.11                             | SolarFluxReborn-1.12.2-12.4.11.jar                          | 9f5e2a811a8332a842b34f6967b7db0ac4f24856 |
       | LC    | supersoundmuffler                 | 1.0.2.10                            | supersoundmuffler-revived_1.12.2_1.0.2.10.jar               | None                                     |
       | LC    | thermalcultivation                | 0.3.6                               | ThermalCultivation-1.12.2-0.3.6.1-universal.jar             | None                                     |
       | LC    | thermalinnovation                 | 0.3.6                               | ThermalInnovation-1.12.2-0.3.6.1-universal.jar              | None                                     |
       | LC    | tickprofiler                      | 1.12-0.0.10                         | TickProfiler-1.12-0.0.10.jar                                | None                                     |
       | LC    | tombmanygraves                    | 1.12-4.2.0                          | TombManyGraves-1.12-4.2.0.jar                               | None                                     |
       | LC    | topaddons                         | 1.12.2-1.13.0                       | topaddons-1.12.2-1.13.0.jar                                 | None                                     |
       | LC    | torchmaster                       | 1.8.5.0                             | torchmaster_1.12.2-1.8.5.0.jar                              | None                                     |
       | LC    | trashslot                         | 8.4.10                              | TrashSlot_1.12.2-8.4.10.jar                                 | None                                     |
       | LC    | ulv_covers                        | 0.2.2                               | ULVCovers-1.12.2-0.2.2.jar                                  | None                                     |
       | LC    | wct                               | 3.12.97                             | WirelessCraftingTerminal-1.12.2-3.12.97.jar                 | 186bc454cd122c9c2f1aa4f95611254bcc543363 |
       | LC    | worleycaves                       | 1.5.2                               | worleycaves-1.12.2-1.5.2.jar                                | None                                     |
       | LC    | xtones                            | 1.2.2                               | Xtones-1.2.2.jar                                            | None                                     |
       | LC    | densemetals                       | 2.0.0                               | DenseMetals-1.12.2-2.0.0.30.jar                             | d476d1b22b218a10d845928d1665d45fce301b27 |
       | LC    | phosphor-lighting                 | 1.12.2-0.2.6                        | phosphor-1.12.2-0.2.6+build50-universal.jar                 | f0387d288626cc2d937daa504e74af570c52a2f1 |
       | LC    | librarianliblate                  | 4.22                                | librarianlib-1.12.2-4.22.jar                                | None                                     |
       
       
       Plugins:
       
       | State | ID          | Version         | Source                                             | Signature |
       |:----- |:----------- |:--------------- |:-------------------------------------------------- |:--------- |
       | LC    | azlink      | 1.2.0           | AzLink.jar                                         | None      |
       | LC    | nucleus     | 2.4.2           | Nucleus-2.4.2-MC1.12.2-SpongeAPI7.3-plugin.jar     | None      |
       | LC    | magibridge  | 2.13            | MagiBridge-API7-2.13.jar                           | None      |
       | LC    | banmanager  | 7.8.0           | BanManagerSponge.jar                               | None      |
       | LC    | epicbanitem | 0.4.1           | EpicBanItem-0.4.1-b343.jar                         | None      |
       | LC    | luckperms   | 5.3.98          | LuckPerms-Sponge-5.3.98.jar                        | None      |
       | LC    | plan        | 5.4 build 1690  | Plan-5.4-build-1690.jar                            | None      |
       | LC    | prism       | 3.0.0           | prism-f456812.jar                                  | None      |
       | LC    | rankupper   | 4.0.5           | RankUpper-4.0.5-b28-Universal.jar                  | None      |
       | LC    | tablist     | 1.0.4           | tablist-sponge-1.0.4.jar                           | None      |
       | LC    | worldedit   | 6.1.10-SNAPSHOT | worldedit-sponge-api7.1.0-6.1.10-SNAPSHOT-dist.jar | None      |
  Loaded coremods (and transformers): RandomPatches (randompatches-1.12.2-1.22.1.10.jar)
                                        com.therandomlabs.randompatches.core.RPTransformer
                                      -Bansoukou- (_bansoukou-4.3.1.jar)
                                        
                                      LibLoader (# LibLoader.jar)
                                        
                                      RBLoadingPlugin (RealBench-1.12.2-1.3.3.jar)
                                        pw.prok.realbench.asm.RBTransformer
                                      LibrarianLib Plugin (librarianlib-1.12.2-4.22.jar)
                                        com.teamwizardry.librarianlib.asm.LibLibTransformer
                                      ForgelinPlugin (Forgelin-1.8.4.jar)
                                        
                                      Do not report to Forge! (If you haven't disabled the FoamFix coremod, try disabling it in the config! Note that this bit of text will still appear.) (foamfix-0.10.14-1.12.2.jar)
                                        pl.asie.foamfix.coremod.FoamFixTransformer
                                      LoliASM (censoredasm5.6.jar)
                                        
                                      CoreMod (TickProfiler-1.12-0.0.10.jar)
                                        
                                      SpongeCoremod (aaa_spongeforge-1.12.2-2838-7.4.8-RC4142.jar)
                                        org.spongepowered.common.launch.transformer.SpongeSuperclassTransformer
                                      GregTechLoadingPlugin (gregtech-1.12.2-2.5.2-beta.jar)
                                        gregtech.asm.GregTechTransformer
                                      JeiUtilitiesLoadingPlugin (JEI-Utilities-1.12.2-0.2.10.jar)
                                        com.github.vfyjxf.jeiutilities.asm.JeiUtilitiesClassTransformer
                                      TickCentral (TickCentral-3.2.jar)
                                        com.github.terminatornl.laggoggles.tickcentral.EventBusTransformer
                                        com.github.terminatornl.tickcentral.asm.BlockTransformer
                                        com.github.terminatornl.tickcentral.asm.ITickableTransformer
                                        com.github.terminatornl.tickcentral.asm.EntityTransformer
                                        com.github.terminatornl.tickcentral.asm.HubAPITransformer
                                      AdvancedRocketryPlugin (AdvancedRocketry-1.12.2-1.7.0-232-universal.jar)
                                        zmaster587.advancedRocketry.asm.ClassTransformer
                                      MixinBooter (!mixinbooter-4.2.jar)
                                        
                                      Inventory Tweaks Coremod (InventoryTweaks-1.63.jar)
                                        invtweaks.forge.asm.ContainerTransformer
                                      EnderCorePlugin (EnderCore-1.12.2-0.5.76-core.jar)
                                        com.enderio.core.common.transform.EnderCoreTransformer
                                        com.enderio.core.common.transform.SimpleMixinPatcher
                                      PhosphorFMLLoadingPlugin (phosphor-1.12.2-0.2.6+build50-universal.jar)
                                        
                                      CreativePatchingLoader (CreativeCore_v1.10.69_mc1.12.2.jar)
                                        
                                      AE2ELCore (appliedenergistics2-rv6-stable-7-extended_life-v0.54.19.jar)
                                        appeng.core.transformer.AE2ELTransformer
                                      LittlePatchingLoader (LittleTiles_v1.5.57_mc1.12.2.jar)
                                        com.creativemd.littletiles.LittleTilesTransformer
  AE2 Version: stable rv6-stable-7-extended_life-v0.54.19 for Forge 14.23.5.2847
  Ender IO: Found the following problem(s) with your installation (That does NOT mean that Ender IO caused the crash or was involved in it in any way. We add this information to help finding common problems, not as an invitation to post any crash you encounter to Ender IO's issue tracker. Always check the stack trace above to see which mod is most likely failing.):
                              * Offline mode for dedicated servers is NOT supported by Ender IO.
                             An unsupported base software is installed: 'sponge'. This is NOT supported.
                             This may (look up the meaning of 'may' in the dictionary if you're not sure what it means) have caused the error. Try reproducing the crash WITHOUT this/these mod(s) before reporting it.
            Authlib is : /C:/Minecraft%20Server/Sponge/Nomifactory/minecraft_server.1.12.2.jar
            
            !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
            !!!You are looking at the diagnostics information, not at the crash.       !!!
            !!!Scroll up until you see the line with '---- Minecraft Crash Report ----'!!!
            !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
  List of loaded APIs: * actuallyadditionsapi (34) from ActuallyAdditions-1.12.2-r152.jar
                       * ae2wtlib|API (1.0.34) from AE2WTLib-1.12.2-1.0.34.jar
                       * appliedenergistics2|API (rv6) from appliedenergistics2-rv6-stable-7-extended_life-v0.54.19.jar
                       * ArmorPlusApi (2.1) from armorplus-1.12.2-11.25.1.62.jar
                       * Base|API (1.0.0) from base-1.12.2-3.14.0.jar
                       * Baubles|API (1.4.0.2) from Baubles-1.12-1.5.2.jar
                       * BetterQuesting|API (3.2) from BetterQuestingUnofficial-4.1.0.jar
                       * BetterQuesting|API2 (3.1) from BetterQuestingUnofficial-4.1.0.jar
                       * BetterWithModsAPI (Beta 0.6) from AppleSkin-mc1.12-1.0.14.jar
                       * chesttransporter|api (2) from ChestTransporter-1.12.2-2.8.8.jar
                       * Chisel-API (0.0.1) from Chisel-MC1.12.2-1.0.2.45.jar
                       * ChiselAPI|Carving (0.0.1) from Chisel-MC1.12.2-1.0.2.45.jar
                       * cofhapi (2.5.0) from CoFHCore-1.12.2-4.6.6.1-universal.jar
                       * CraftingTweaks|API (4.1) from CraftingTweaks_1.12.2-9.0.1.jar
                       * DraconicEvolution|API (1.3) from Draconic-Evolution-1.12.2-2.3.28.354-universal.jar
                       * enderioapi (4.0.0) from EnderIO-1.12.2-5.3.70.jar
                       * enderioapi|addon (4.0.0) from EnderIO-1.12.2-5.3.70.jar
                       * enderioapi|capacitor (4.0.0) from EnderIO-1.12.2-5.3.70.jar
                       * enderioapi|conduits (4.0.0) from EnderIO-1.12.2-5.3.70.jar
                       * enderioapi|farm (4.0.0) from EnderIO-1.12.2-5.3.70.jar
                       * enderioapi|redstone (4.0.0) from EnderIO-1.12.2-5.3.70.jar
                       * enderioapi|teleport (4.0.0) from EnderIO-1.12.2-5.3.70.jar
                       * enderioapi|tools (4.0.0) from EnderIO-1.12.2-5.3.70.jar
                       * enderioapi|upgrades (4.0.0) from EnderIO-1.12.2-5.3.70.jar
                       * jeresources|API (0.9.2.60) from JustEnoughResources-1.12.2-0.9.2.60.jar
                       * journeymap|client-api (1.4) from journeymap-1.12.2-5.7.1.jar
                       * journeymap|client-api-display (1.4) from journeymap-1.12.2-5.7.1.jar
                       * journeymap|client-api-event (1.4) from journeymap-1.12.2-5.7.1.jar
                       * journeymap|client-api-model (1.4) from journeymap-1.12.2-5.7.1.jar
                       * journeymap|client-api-util (1.4) from journeymap-1.12.2-5.7.1.jar
                       * JustEnoughItemsAPI (4.13.0) from HadEnoughItems_1.12.2-4.23.0.jar
                       * Open Glider|API (0.1) from OpenGlider-1.12.1-1.1.0.jar
                       * PatchouliAPI (6) from Patchouli-1.0-23.6.jar
                       * projectred|api (2.1) from ProjectRed-1.12.2-4.9.4.120-Base.jar
                       * redstonefluxapi (2.1.1) from RedstoneFlux-1.12-2.1.1.1-universal.jar
                       * StorageDrawersAPI (2.1.0) from StorageDrawers-1.12.2-5.4.2.jar
                       * StorageDrawersAPI|event (2.1.0) from StorageDrawers-1.12.2-5.4.2.jar
                       * StorageDrawersAPI|registry (2.1.0) from StorageDrawers-1.12.2-5.4.2.jar
                       * StorageDrawersAPI|render (2.1.0) from StorageDrawers-1.12.2-5.4.2.jar
                       * StorageDrawersAPI|storage (2.1.0) from StorageDrawers-1.12.2-5.4.2.jar
                       * StorageDrawersAPI|storage-attribute (2.1.0) from StorageDrawers-1.12.2-5.4.2.jar
                       * theoneprobe_api (1.4.4) from theoneprobe-1.12-1.4.28.jar
                       * wct|api (1.1) from WirelessCraftingTerminal-1.12.2-3.12.97.jar
  Suspected Mods: GregTech (gregtech), Forge Mod Loader (FML), Minecraft Forge (forge)
  Profiler Position: N/A (disabled)
  Is Modded: Definitely; Server brand changed to 'fml,forge,sponge'
  Type: Dedicated Server (map_server.txt)
message.txt
48 KB

### Additional Information

_No response_
```

--- a/src/main/java/gregtech/api/damagesources/DamageSources.java
+++ b/src/main/java/gregtech/api/damagesources/DamageSources.java
@@ -54,7 +54,7 @@ public class DamageSources {
         ItemStack stack = source != null ? source.getHeldItemMainhand() : ItemStack.EMPTY;
         if (!stack.isEmpty() && stack.getItem() instanceof IGTTool) {
             IGTTool tool = (IGTTool) stack.getItem();
-            return new DamageSourceTool("player", source, String.format("death.attack.%s", tool.getId()));
+            return new DamageSourceTool("player", source, String.format("death.attack.%s", tool.getToolId()));
         }
         return new EntityDamageSource("player", source);
     }
@@ -65,7 +65,7 @@ public class DamageSources {
         ItemStack stack = source != null ? source.getItemStackFromSlot(EntityEquipmentSlot.MAINHAND) : ItemStack.EMPTY;
         if (!stack.isEmpty() && stack.getItem() instanceof IGTTool) {
             IGTTool tool = (IGTTool) stack.getItem();
-            return new DamageSourceTool("mob", source, String.format("death.attack.%s", tool.getId()));
+            return new DamageSourceTool("mob", source, String.format("death.attack.%s", tool.getToolId()));
         }
         return new EntityDamageSource("mob", source);
     }

--- a/src/main/java/gregtech/api/items/toolitem/IGTTool.java
+++ b/src/main/java/gregtech/api/items/toolitem/IGTTool.java
@@ -100,7 +100,7 @@ public interface IGTTool extends ItemUIFactory, IAEWrench, IToolWrench, IToolHam
     /**
      * @return the name of the tool
      */
-    String getId();
+    String getToolId();
 
     boolean isElectric();
 
@@ -721,7 +721,7 @@ public interface IGTTool extends ItemUIFactory, IAEWrench, IToolWrench, IToolHam
         toolStats.getBehaviors().forEach(behavior -> behavior.addInformation(stack, world, tooltip, flag));
 
         // unique tooltip
-        String uniqueTooltip = "item.gt.tool." + getId() + ".tooltip";
+        String uniqueTooltip = "item.gt.tool." + getToolId() + ".tooltip";
         if (I18n.hasKey(uniqueTooltip)) {
             tooltip.add("");
             tooltip.add(I18n.format(uniqueTooltip));
@@ -820,7 +820,7 @@ public interface IGTTool extends ItemUIFactory, IAEWrench, IToolWrench, IToolHam
 
     @SideOnly(Side.CLIENT)
     default String getModelPath() {
-        return getDomain() + ":" + "tools/" + getId();
+        return getDomain() + ":" + "tools/" + getToolId();
     }
 
     @SideOnly(Side.CLIENT)

--- a/src/main/java/gregtech/api/items/toolitem/ItemGTAxe.java
+++ b/src/main/java/gregtech/api/items/toolitem/ItemGTAxe.java
@@ -66,7 +66,7 @@ public class ItemGTAxe extends ItemAxe implements IGTTool {
     }
 
     @Override
-    public String getId() {
+    public String getToolId() {
         return id;
     }
 

--- a/src/main/java/gregtech/api/items/toolitem/ItemGTHoe.java
+++ b/src/main/java/gregtech/api/items/toolitem/ItemGTHoe.java
@@ -66,7 +66,7 @@ public class ItemGTHoe extends ItemHoe implements IGTTool {
     }
 
     @Override
-    public String getId() {
+    public String getToolId() {
         return id;
     }
 

--- a/src/main/java/gregtech/api/items/toolitem/ItemGTSword.java
+++ b/src/main/java/gregtech/api/items/toolitem/ItemGTSword.java
@@ -70,7 +70,7 @@ public class ItemGTSword extends ItemSword implements IGTTool {
     }
 
     @Override
-    public String getId() {
+    public String getToolId() {
         return id;
     }
 

--- a/src/main/java/gregtech/api/items/toolitem/ItemGTTool.java
+++ b/src/main/java/gregtech/api/items/toolitem/ItemGTTool.java
@@ -72,7 +72,7 @@ public class ItemGTTool extends ItemTool implements IGTTool {
     }
 
     @Override
-    public String getId() {
+    public String getToolId() {
         return id;
     }
 

--- a/src/main/java/gregtech/loaders/recipe/handlers/ToolRecipeHandler.java
+++ b/src/main/java/gregtech/loaders/recipe/handlers/ToolRecipeHandler.java
@@ -16,7 +16,6 @@ import gregtech.api.unification.material.properties.PropertyKey;
 import gregtech.api.unification.material.properties.ToolProperty;
 import gregtech.api.unification.ore.OrePrefix;
 import gregtech.api.unification.stack.UnificationEntry;
-import gregtech.common.ConfigHolder;
 import gregtech.common.crafting.ToolHeadReplaceRecipe;
 import gregtech.common.items.MetaItems;
 import gregtech.common.items.ToolItems;
@@ -296,7 +295,7 @@ public class ToolRecipeHandler {
             ItemStack powerUnitStack = powerUnitItems.get(tier).getStackForm();
             IElectricItem powerUnit = powerUnitStack.getCapability(GregtechCapabilities.CAPABILITY_ELECTRIC_ITEM, null);
             ItemStack tool = toolItem.get(material, 0, powerUnit.getMaxCharge());
-            ModHandler.addShapedEnergyTransferRecipe(String.format("%s_%s", toolItem.getId(), material),
+            ModHandler.addShapedEnergyTransferRecipe(String.format("%s_%s", toolItem.getToolId(), material),
                     tool,
                     Ingredient.fromStacks(powerUnitStack), true, true,
                     "wHd", " U ",
@@ -307,10 +306,10 @@ public class ToolRecipeHandler {
 
     public static void addToolRecipe(@Nonnull Material material, @Nonnull IGTTool tool, boolean mirrored, Object... recipe) {
         if (mirrored) {
-            ModHandler.addMirroredShapedRecipe(String.format("%s_%s", tool.getId(), material),
+            ModHandler.addMirroredShapedRecipe(String.format("%s_%s", tool.getToolId(), material),
                     tool.get(material), recipe);
         } else {
-            ModHandler.addShapedRecipe(String.format("%s_%s", tool.getId(), material),
+            ModHandler.addShapedRecipe(String.format("%s_%s", tool.getToolId(), material),
                     tool.get(material), recipe);
         }
     }

